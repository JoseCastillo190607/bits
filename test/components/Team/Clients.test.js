import { render, cleanup, fireEvent } from '@testing-library/react';
import Clients from '../../../components/Team/Clients';
import SedeClientModal from '../../../components/Team/Modals/AddRegisterModal';
import TeamContext from '../../../context/ModalContext/ModalContext';

afterEach(() => {
    cleanup();
});

const setup = () => {
    const utils = render(<Clients />);
    const input = utils.getByTestId('newCliente');
    return {
        input,
        ...utils,
    }
}

describe('Testing <Clients />', () => {
    test('should render Clients', () => {
        const { input } = setup();
        fireEvent.change(input, { target: { value: 'new cliente' } });
        if (input.value === '') console.log('sin texto, no deberia hacer get');
        else expect(input).toMatchSnapshot();
    });
});

describe('Testing <SedeClientModal />', () => {
    test('should render modal', async () => {
        const Component = render(
            <TeamContext.Provider value={{ state: { open: true, value: 'testiando cliente', _id: 'id de testiando cliente' }, dispatch: jest.fn() }}>
                <SedeClientModal type="Cliente" func={() => { }} />
            </TeamContext.Provider>
        );
        expect(Component.getByTestId('TitleModal').textContent).toEqual("Editando Cliente");
        expect(Component.getByTestId('form-dialog-text-type')).toMatchSnapshot();
    });
});