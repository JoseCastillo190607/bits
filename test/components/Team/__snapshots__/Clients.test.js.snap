// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Clients /> should render Clients 1`] = `
<input
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
  data-testid="newCliente"
  placeholder="Nombre de cliente"
  type="text"
  value="new cliente"
/>
`;

exports[`Testing <SedeClientModal /> should render modal 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
  data-testid="form-dialog-text-type"
  placeholder="Cliente"
  required=""
  type="text"
  value="testiando cliente"
/>
`;
